agent/configmgr/fleet/auth.go

```diff
@@ -0,0 +1,125 @@
+package fleet
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log/slog"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+)
+
+// AuthTokenManager manages auth tokens
+type AuthTokenManager struct {
+	logger *slog.Logger
+}
+
+// NewAuthTokenManager creates a new AuthTokenManager
+func NewAuthTokenManager(logger *slog.Logger) *AuthTokenManager {
+	return &AuthTokenManager{
+		logger: logger,
+	}
+}
+
+// TokenResponse is the response from the auth token endpoint
+type TokenResponse struct {
+	AccessToken string `json:"access_token"`
+	MQTTURL     string `json:"mqtt_url"`
+	ExpiresIn   int    `json:"expires_in"`
+}
+
+// GetToken gets an auth token from the auth token endpoint
+func (fleetManager *AuthTokenManager) GetToken(ctx context.Context, tokenURL string, skipTLS bool, timeout time.Duration, clientID string, clientSecret string) (*TokenResponse, error) {
+	// Input validation
+	if tokenURL == "" {
+		return nil, fmt.Errorf("token URL cannot be empty")
+	}
+	if clientID == "" {
+		return nil, fmt.Errorf("client ID cannot be empty")
+	}
+	if clientSecret == "" {
+		return nil, fmt.Errorf("client secret cannot be empty")
+	}
+
+	fleetManager.logger.Debug("requesting access token", "token_url", tokenURL, "client_id", clientID)
+
+	scopes := []string{
+		"orb.mqtt:agent",
+		"orb.mqtt:group",
+	}
+
+	data := url.Values{}
+	data.Set("grant_type", "client_credentials")
+	data.Set("scope", strings.Join(scopes, " "))
+	data.Set("client_id", clientID)
+	data.Set("client_secret", clientSecret)
+	data.Set("audience", "orb")
+
+	fleetManager.logger.Debug("sending token request", "url", tokenURL, "data", data, "client_id", clientID) //, "client_secret", clientSecret)
+
+	req, err := http.NewRequest("POST", tokenURL, bytes.NewBufferString(data.Encode()))
+	if err != nil {
+		fleetManager.logger.Error("failed to create token request", "error", err, "token_url", tokenURL)
```

Comment:
Not important, but something to consider long term.

Perhaps it would be best to use log statements with context (separate PR). I think the use case this could serve is adding metrics to log statements or alerting on error log statements. When those metrics/alerts are created from log statements, they would have access to the incoming request's context, which could contain useful information.

```suggestion
		fleetManager.logger.ErrorContext(ctx, "failed to create token request", "error", err, "token_url", tokenURL)
```

Reply:
Good suggestion. I didn't know about this. I'll look to add it in where I can in a future PR.